### ISSUE TITLE
Dispatch "Use custom middleware for async actions" is confusing and should be updated

### ISSUE THREAD

**Issue by markerikson** [reduxjs/redux#4028]
The core `dispatch` function currently starts with:

```js
  function dispatch(action) {
    if (!isPlainObject(action)) {
      throw new Error(
        'Actions must be plain objects. ' +
          'Use custom middleware for async actions.'
      )
    }
```

Frankly, I've always disliked the "async actions" term, which is part of why I avoided it completely in the new tutorials.  The phrase seems confusing anyway.

On top of that, this doesn't actually tell you _what_ was dispatched, which might be helpful for knowing what the actual problem is - is it a Promise, a function, `undefined`, or something else?

We should reword this to drop the "async actions" phrase, and also have it log the received value.  

I'm wondering if there's a good way to log the value itself.  I know that over in React-Redux, we have a `stringifyComponent` function that tries to provide more details and falls back if it can't be stringified:

```js
const stringifyComponent = (Comp) => {
  try {
    return JSON.stringify(Comp)
  } catch (err) {
    return String(Comp)
  }
}
```

maybe something like that would be relevant here?  

This should be doable as a patch release for 4.x, so presumably 4.0.6.

**Comment by timdorr** [comment]
I think the only issue is if the object is really large, then the `JSON.stringify` output would be as well. I don't know if there's a good workaround so we don't end up spamming console log with a bunch of junk.

**Comment by markerikson** [comment]
slice the string to a max of 100 characters or something? :)

**Comment by markerikson** [comment]
I'm playing with inlining a cut-down version of https://github.com/jonschlinkert/kind-of in dev mode. That at least prints out `'Promise'`, `'Date'`, etc, instead of just `'object'`, and then it falls back to `typeof` in prod.

**Comment by markerikson** [comment]
Live now in https://github.com/reduxjs/redux/releases/tag/v4.1.0-alpha.0

**Comment by markerikson** [comment]
Could still consider logging the actual values rather than the name of the type, if we think that would be beneficial.
